design/arp-load-sharing.md

```diff
@@ -0,0 +1,41 @@
+# ARP Load Sharing
+
+## Summary
+
+MetalLB in layer 2 mode does not support load balancing. See [here](https://metallb.universe.tf/concepts/layer2/).
+However, in some cases, L2 is the only option, yet we still require a load balancing solution which is internal to our infrastructure.
+
+A method that can achieve that is known as "ARP Load Sharing". See for example [here](https://docs.paloaltonetworks.com/pan-os/10-0/pan-os-admin/high-availability/ha-concepts/arp-load-sharing). This technique involves fiddling with how ARP replies are sent, which raises the concern that it might get blocked by the network switch or security policy since it seems like an [ARP Spoofing](https://en.wikipedia.org/wiki/ARP_spoofing) attack. See more information about "does arp spoofing work on all lans" [here](https://security.stackexchange.com/questions/133784/does-arp-spoofing-work-on-all-lans).
+
+This document proposes adding an option to MetalLB to support "ARP Load Sharing" mode.
+
+## Motivation
+
+When BGP is not available, in order to support load balancing and HA for a Kubernetes Service, we need to create as many Services as there are nodes, so that each node will be an owner of one VIP and that node will be able to get traffic directly from clients. The VIP is needed to support HA so that on node failure the VIP can move to another node. However, even with a service per node, we still need to add a higher level load balancer, which will balance those VIPs for the clients (for example using switches that can perform L3-routing, DNS round-robin, etc.).
+
+We want to simplify this setup, and have an option to "serve" a VIP from all nodes.
```

Comment:
What is the use case to do this? I mean, why is that serving from one node (and fast failover if that node fails) is problematic? Is it because your ingress is limited to the ingress of a single node? Is there some other reason?

Can we state what is the underlying problem we want to solve by adding arp load balancing?